04-store-api/starter/controllers/products.js

```diff
@@ -0,0 +1,102 @@
+const Product = require('../models/product')
+
+
+const getAllProductsStatic = async (req, res) => {
+   const search = 'ab'
+   const products = await Product.find({
+         price: {
+            $gt: 30
+         }
+      })
+      .sort('price')
+      .select('name price')
+
+   res.status(200).json({
+      products,
+      nbHits: products.length
+   })
+}
+
+const getAllProducts = async (req, res) => {
+   const {
+      featured,
+      company,
+      name,
+      sort,
+      fields,
+      numericFilters
+   } = req.query
+   const queryObject = {}
+
+   if (featured) {
+      queryObject.featured = featured === 'true' ? true : false
+   }
+
+   if (company) {
+      queryObject.company = company
+   }
+
+   if (name) {
+      queryObject.name = {
+         $regex: name,
+         $options: 'i'
+      }
+   }
+
+   if (numericFilters) {
+      const operatorMap = {
+         '>': '$gt',
+         '>=': '$gte',
+         '=': '$eq',
+         '<': '$lt',
+         '<=': '$lte',
+      }
+
+      const regEx = /\b(<|>|>=|=|<|<=)\b/g
+      let filters = numericFilters.replace(regEx,
+         (match) => `-${operatorMap[match]}-`
+      )
+      const options = ['price', 'rating']
+      filters = filters.split(',').forEach((item) => {
+         const [field, operator, value] = item.split('-')
+         if (options.includes(field)) {
+            queryObject[field] = {
+               [operator]: Number(value)
+            }
+         }
+      })
+
+
+   }
+   console.log(queryObject)
+   let result = Product.find(queryObject)
+   // sort
+   if (sort) {
+      const sortList = sort.split(',').join(' ');
+      result = result.sort(sortList)
+   } else {
+      result = result.sort('createAt')
```

Comment:
This should be `createdAt` it's missing the 'd' , so currently it won't correctly sort the products since the mongo schema for products doesn't have a `createAt` field.

Reply:
Hello Akos, I have edited the line 78. Thank you very much !

Reply:
:+1:  looks great, ready to merge!